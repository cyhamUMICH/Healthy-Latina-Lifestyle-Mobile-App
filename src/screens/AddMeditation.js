import React, {useState, useEffect} from 'react';
import { styles } from '../styles/Styles';
import { colors } from '../styles/Colors';
import { View, KeyboardAvoidingView, ScrollView, Text, TextInput, Alert } from 'react-native';
import { Button, ButtonGroup, CheckBox } from 'react-native-elements';
import { GetTopics } from '../components/GetTopics';
import TopicButtons from '../components/TopicButtons';
import UploadImage from '../components/UploadImage';
import UploadAudio from '../components/UploadAudio';
import LoadingSpinner from '../components/LoadingSpinner';
import * as firebase from 'firebase/app';
import "firebase/firestore";
import "firebase/storage";

const AddMeditation = ({route}) => {
  const {topics, navigation} = route.params;

  const [title, setTitle] = useState();
  const [description, setDescription] = useState();
  const languageOptions = [{label: 'English', value: 'EN'}, {label: 'Spanish', value: 'ES'}];
  const [selectedLanguage, setSelectedLanguage] = useState();
  const difficultyOptions = [{label: 'Beginner', value: 'B'}, {label: 'Advanced', value: 'A'}];
  const [selectedDifficulty, setSelectedDifficulty] = useState();
  const [isTopicsLoaded, setIsTopicsLoaded] = useState(false);
  const [topicsList, setTopicsList] = useState([]);
  const [selectedTopics, setSelectedTopics] = useState([]);
  const [image, setImage] = useState(null);
  const [audio, setAudio] = useState(null);
  const [duration, setDuration] = useState();
  const [cost, setCost] = useState();
  const [isFeatured, setIsFeatured] = useState(false);

  const [isUploadInProgress, setIsUploadInProgress] = useState(false);

  const baseImagePath = "meditations/images/";
  const baseAudioPath = "meditations/audios/";

  useEffect(() => {
    if (topicsList.length === 0) {
      if (topics != null) {
        setTopicsList(topics);
        setIsTopicsLoaded(true);
      }
      else {
        GetTopics(setIsTopicsLoaded, setTopicsList);
      }
    }
  }, []);

  const formComplete = () => {
    return (title && description && selectedLanguage && selectedDifficulty && selectedTopics && image && audio
      && duration && cost);
  };

  const validateDuration = () => {
    return duration && !isNaN(parseInt(duration));
  }

  const validateCost = () => {
    return cost && !isNaN(parseFloat(cost));
  }

  const submit = async () => {

    if (formComplete() && validateDuration() && validateCost()) 
    {
      const dbh = firebase.firestore();
      const docRef = await dbh.collection("meditations").add({
        title: title,
        description: description,
        language: selectedLanguage,
        difficulty: selectedDifficulty,
        topics: selectedTopics,
        duration: parseInt(duration),
        cost: parseFloat(cost),
        featured: isFeatured,
        dateAdded: new Date()
      });
      
      await docRef.update({
        imagePath: baseImagePath + docRef.id + "__" + image.filename,
        audioPath: baseAudioPath + docRef.id + "__" + audio.name
      });

      const imageLoc = firebase.storage().ref().child(baseImagePath.concat(docRef.id).concat("__").concat(image.filename));
      // Code from: https://medium.com/@ericmorgan1/upload-images-to-firebase-in-expo-c4a7d4c46d06
      const imageResponse = await fetch(image.uri);
      const imageBlob = await imageResponse.blob();
      setIsUploadInProgress(true);
      let uploadImageStatus = imageLoc.put(imageBlob);

      // Code from: https://firebase.google.com/docs/reference/js/firebase.storage.UploadTask#on
      uploadImageStatus.on(firebase.storage.TaskEvent.STATE_CHANGED, {
        'complete': async function() {
          const audioLoc = firebase.storage().ref().child(baseAudioPath.concat(docRef.id).concat("_").concat(audio.name));
          // Code from: https://medium.com/@ericmorgan1/upload-images-to-firebase-in-expo-c4a7d4c46d06
          const audioResponse = await fetch(audio.uri);
          const audioBlob = await audioResponse.blob();
          let uploadAudioStatus = audioLoc.put(audioBlob);
          uploadAudioStatus.on(firebase.storage.TaskEvent.STATE_CHANGED, {
            'complete': function() {
              setIsUploadInProgress(false);
              navigation.goBack();
              navigation.replace("MeditationList");
            },
            'error': function() {
              Alert.alert(
                "Error Adding Meditation",
                "There was an error when uploading the audio.",
                [
                  {text: "OK"}
                ]
              );
            }
          });
        },
        'error': function() {
          Alert.alert(
            "Error Adding Meditation",
            "There was an error when uploading the image.",
            [
              {text: "OK"}
            ]
          );
        }
      });
    }
    else
    {
      Alert.alert(
        "Error Adding Meditation",
        "Please complete all of the fields before submitting, and check the format of the duration and cost fields.",
        [
          {text: "OK"}
        ]
      );
    }
  };

  return(
    <View style={styles.app}>
      {
        isTopicsLoaded && !isUploadInProgress ?
          <View style={styles.fullWidthWindow}>
            <Text style={styles.contentTitle}>Add Meditation</Text>
              <View style={styles.flexContainer}>
                <KeyboardAvoidingView 
                behavior="padding"
                style={styles.inputView}>
                <ScrollView>
                  <TextInput
                    style={styles.inputText} textAlign="center"
                    placeholder="title" placeholderTextColor={colors.text} 
                    onChangeText={input => setTitle(input)} />
                  <TextInput 
                    style={styles.inputText} textAlign="left"
                    placeholder="description" placeholderTextColor={colors.text}
                    multiline textAlignVertical="top" numberOfLines={6} 
                    onChangeText={input => setDescription(input)} />
                  <Text style={styles.cardTitle}>Language</Text>
                  <ButtonGroup
                    textStyle={styles.buttonGroupText}
                    buttonStyle={styles.buttonGroup}
                    selectedButtonStyle={styles.buttonGroupSelected}
                    onPress={(selectedIndex) => setSelectedLanguage(languageOptions[selectedIndex].value)}
                    selectedIndex={languageOptions.findIndex((lang => lang.value == selectedLanguage))}
                    buttons={languageOptions.map((lang) => lang.label)}
                  />
                  <Text style={styles.cardTitle}>Difficulty</Text>
                  <ButtonGroup
                    textStyle={styles.buttonGroupText}
                    buttonStyle={styles.buttonGroup}
                    selectedButtonStyle={styles.buttonGroupSelected}
                    onPress={(selectedIndex) => setSelectedDifficulty(difficultyOptions[selectedIndex].value)}
                    selectedIndex={difficultyOptions.findIndex((lang => lang.value == selectedDifficulty))}
                    buttons={difficultyOptions.map((lang) => lang.label)}
                  />
                  <Text style={styles.cardTitle}>Topics (Select All)</Text>
                  <TopicButtons 
                    topicsList={topicsList} 
                    topics={selectedTopics} 
                    topicsFunction={setSelectedTopics} />
                  <UploadImage image={image} setImage={setImage} />
                  <UploadAudio audio={audio} setAudio={setAudio} />
                  <TextInput
                    style={styles.inputText} textAlign="center"
                    placeholder="duration (seconds)" placeholderTextColor={colors.text} 
                    onChangeText={input => setDuration(input)} />
                  <TextInput
                    style={styles.inputText} textAlign="center"
                    placeholder="cost" placeholderTextColor={colors.text} 
                    onChangeText={input => setCost(input)} />
                  <CheckBox
                    title="Featured"
                    checkedIcon='check-square'
                    uncheckedIcon='square'
                    checked={isFeatured}
                    checkedColor={colors.accentBackground}
                    uncheckedColor={colors.uncheckedColor}
                    containerStyle={styles.checkBoxContainer}
                    textStyle={styles.checkBoxLabel}
                    onPress={() => setIsFeatured(!isFeatured)}
                  />
                  <Button 
                    title="Submit" 
                    buttonStyle={styles.button}
                    titleStyle={styles.buttonText}
                    onPress={submit} />
                </ScrollView>
              </KeyboardAvoidingView>
            </View>
          </View>
        : <LoadingSpinner />
      }
    </View>
  );
};

export default AddMeditation;