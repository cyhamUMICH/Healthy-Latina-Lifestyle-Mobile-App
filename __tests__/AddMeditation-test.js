// Code from: https://github.com/byCedric/expo-guide-ci/blob/master/__tests__/App-test.js
//      and https://jestjs.io/docs/tutorial-react-native
import React from 'react';
import AddMeditation from '../src/screens/AddMeditation';
import renderer from 'react-test-renderer';

const mockedParams = {
  route: { 
    params: {
      topics: [],
      // https://stackoverflow.com/questions/52569447/how-to-mock-react-navigations-navigation-prop-for-unit-tests-with-typescript-in
      navigation: {
        goBack: jest.fn(),
        replace: jest.fn()
      }
    }
  }
};

test('Renders Add Meditation With the Loading Screen', async () => {
  const tree = renderer.create(<AddMeditation {...mockedParams} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders Add Meditation Without the Loading Screen', async () => {
  const tree = renderer.create(<AddMeditation {...mockedParams} skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});