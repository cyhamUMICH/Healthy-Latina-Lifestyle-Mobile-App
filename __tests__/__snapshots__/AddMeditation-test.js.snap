// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Add Meditation With the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#00A89D"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 2.5,
            },
          ],
        }
      }
    />
  </View>
</View>
`;

exports[`Renders Add Meditation Without the Loading Screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": "2%",
        "paddingTop": "2%",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 28,
          "textAlign": "center",
        }
      }
    >
      Add Meditation
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "marginVertical": "2%",
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <RCTScrollView>
          <View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="title"
              placeholderTextColor="#000"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#00A89D",
                  "borderRadius": 30,
                  "marginBottom": "2%",
                  "paddingHorizontal": "5%",
                  "paddingVertical": "2%",
                }
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
            <TextInput
              allowFontScaling={true}
              multiline={true}
              numberOfLines={6}
              onChangeText={[Function]}
              placeholder="description"
              placeholderTextColor="#000"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#00A89D",
                  "borderRadius": 30,
                  "marginBottom": "2%",
                  "paddingHorizontal": "5%",
                  "paddingVertical": "2%",
                }
              }
              textAlign="left"
              textAlignVertical="top"
              underlineColorAndroid="transparent"
            />
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "lineHeight": 30,
                  "textAlign": "left",
                }
              }
            >
              Language
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#e3e3e3",
                  "borderRadius": 3,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": 40,
                  "marginHorizontal": 10,
                  "marginVertical": 5,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRightColor": "#bdc6cf",
                    "borderRightWidth": 1,
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                  testID="buttonGroupItem"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#00A89D",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000",
                          "fontSize": 20,
                          "fontWeight": "500",
                        }
                      }
                      testID="buttonGroupItemText"
                    >
                      English
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                  testID="buttonGroupItem"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#00A89D",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000",
                          "fontSize": 20,
                          "fontWeight": "500",
                        }
                      }
                      testID="buttonGroupItemText"
                    >
                      Spanish
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "lineHeight": 30,
                  "textAlign": "left",
                }
              }
            >
              Difficulty
            </Text>
            <View
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderColor": "#e3e3e3",
                  "borderRadius": 3,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "height": 40,
                  "marginHorizontal": 10,
                  "marginVertical": 5,
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderRightColor": "#bdc6cf",
                    "borderRightWidth": 1,
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                  testID="buttonGroupItem"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#00A89D",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000",
                          "fontSize": 20,
                          "fontWeight": "500",
                        }
                      }
                      testID="buttonGroupItemText"
                    >
                      Beginner
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "opacity": 1,
                    }
                  }
                  testID="buttonGroupItem"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#00A89D",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000",
                          "fontSize": 20,
                          "fontWeight": "500",
                        }
                      }
                      testID="buttonGroupItemText"
                    >
                      Advanced
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "lineHeight": 30,
                  "textAlign": "left",
                }
              }
            >
              Topics (Select All)
            </Text>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-evenly",
                  "marginVertical": "1%",
                }
              }
            />
            <View>
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 3,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 3,
                    },
                    undefined,
                    false,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#00A89D",
                        "borderColor": "#2089dc",
                        "borderRadius": 3,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "marginBottom": "2%",
                        "padding": 8,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000",
                          "fontSize": 18,
                          "paddingVertical": 1,
                          "textAlign": "center",
                        }
                      }
                    >
                      Upload Image
                    </Text>
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "marginLeft": "5%",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View>
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 3,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 3,
                    },
                    undefined,
                    false,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#00A89D",
                        "borderColor": "#2089dc",
                        "borderRadius": 3,
                        "borderWidth": 0,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "marginBottom": "2%",
                        "padding": 8,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000",
                          "fontSize": 18,
                          "paddingVertical": 1,
                          "textAlign": "center",
                        }
                      }
                    >
                      Upload Audio
                    </Text>
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "marginLeft": "5%",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="duration (seconds)"
              placeholderTextColor="#000"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#00A89D",
                  "borderRadius": 30,
                  "marginBottom": "2%",
                  "paddingHorizontal": "5%",
                  "paddingVertical": "2%",
                }
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
            <TextInput
              allowFontScaling={true}
              onChangeText={[Function]}
              placeholder="cost"
              placeholderTextColor="#000"
              rejectResponderTermination={true}
              style={
                Object {
                  "backgroundColor": "#00A89D",
                  "borderRadius": 30,
                  "marginBottom": "2%",
                  "paddingHorizontal": "5%",
                  "paddingVertical": "2%",
                }
              }
              textAlign="center"
              underlineColorAndroid="transparent"
            />
            <View
              accessibilityRole="checkbox"
              accessibilityState={
                Object {
                  "checked": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "transparent",
                  "borderRadius": 3,
                  "borderWidth": 1,
                  "margin": 5,
                  "marginBottom": "2%",
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 0,
                  "opacity": 1,
                  "padding": 0,
                }
              }
              testID="checkbox"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text />
                <Text
                  style={
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                      "fontWeight": "normal",
                      "marginLeft": "2%",
                      "marginRight": 10,
                    }
                  }
                  testID="checkboxTitle"
                >
                  Featured
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderRadius": 3,
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 3,
                  },
                  undefined,
                  false,
                ]
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "busy": false,
                    "disabled": false,
                  }
                }
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#00A89D",
                      "borderColor": "#2089dc",
                      "borderRadius": 3,
                      "borderWidth": 0,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginBottom": "2%",
                      "padding": 8,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#000",
                        "fontSize": 18,
                        "paddingVertical": 1,
                        "textAlign": "center",
                      }
                    }
                  >
                    Submit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</View>
`;
